### PR TITLE
HTCondor Job Queue High Availability

### PR DESCRIPTION
**Pull request by tpdownes**
This PR adds support for [Job Queue High Availability](https://htcondor.readthedocs.io/en/latest/admin-manual/high-availability.html#high-availability-of-the-job-queue) to the htcondor-configure module. During development of the module, we raised 2 issues to the HTCondor team. These issues have been resolved and will be made available in forthcoming releases, identified in our documentation:

- [HTCONDOR-1590](https://opensciencegrid.atlassian.net/jira/software/c/projects/HTCONDOR/issues/HTCONDOR-1590)
- [HTCONDOR-1594](https://opensciencegrid.atlassian.net/jira/software/c/projects/HTCONDOR/issues/HTCONDOR-1594)

Accordingly the feature is marked experimental.

### Submission Checklist

* [X] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [X] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [X] Is unit test coverage still above 80%?
* [X] Have you updated all applicable documentation?
* [X] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/scheduler/htcondor-configure/README.md
Comment (tpdownes) [inline community/modules/scheduler/htcondor-configure/README.md:75]: FYI: this text appears above the YAML snippet now. So this is removing duplicated text.

File: community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml
Comment (tpdownes) [inline community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml:176]: FYI: this SystemD override can be removed once HTCondor releases the fixes. it is harmless once they do.

TL;DR: it tells SystemD to start HTCondor only after any fstab entries necessary to access that the `{{ spool_dir }}` have been mounted.

File: community/modules/scheduler/htcondor-configure/README.md
Comment (nick-stroud) [inline community/modules/scheduler/htcondor-configure/README.md:82]: Subnetwork is also different between the two `vm-instances` defined below. Not sure if this is a typo or correct but either way it seems to conflict with the way the sentence is written: `should differ only by`.

Comment (nick-stroud) [inline community/modules/scheduler/htcondor-configure/README.md:82]: Just from naive inspection it would seem both of these VMs would inherit the same `zone` from top level vars.  This seems counter to being resilient to "zonal failures". What am I missing here?

File: community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml
Comment (nick-stroud) [inline community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml:197]: Should `schedd_ha_config_file` be defined at the top level `vars` block? Where is it coming from?

File: community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml
Comment (nick-stroud) [reply community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml:176]: I would add a `# TODO` in the code. If you disagree with that then perhaps a task scheduled for future. I am just betting we will not come back to this comment.

File: community/modules/scheduler/htcondor-configure/README.md
Comment (nick-stroud) [inline community/modules/scheduler/htcondor-configure/README.md:80]: Here I am interpreting "apply the startup script in the example" as "have two VMs `use` the startup script generated by `htcondor_central_manager_startup` in the example above. 

My critique here being:
- The use of `apply` feels overloaded and maybe tying to the `use` keyword would avoid confusion.
- The reference to above example could be more explicit.

File: community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml:180]: I added a comment in the code explaining when it can be removed and also created a task. It is front and center in my mind but it is good to leave a trail of crumbs.

File: community/modules/scheduler/htcondor-configure/README.md
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/README.md]: I've updated both the blueprint setting and surrounding text.

File: community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml:201]: Good catch. This got lost in a rebase to prepare this PR. I've fixed it and re-ran a blueprint that I've been using for manual testing. While testing that, I found that another setting ([SCHEDD_NAME](https://htcondor.readthedocs.io/en/latest/admin-manual/high-availability.html#working-with-remote-job-submission)) can be important for use cases where jobs are submitted remotely. I have added that setting.

File: community/modules/scheduler/htcondor-configure/README.md
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/README.md]: I've modified the surround text to note all the changes and massaged the verbs to something less "loaded"

File: community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/files/htcondor_configure.yml:201]: I have created a task to add the HA blueprint to the repo and an integration test. I believe this should wait until the 2 bugs below land in a feature release (should be 10.4 in approximately early April)

- [HTCONDOR-1590](https://opensciencegrid.atlassian.net/jira/software/c/projects/HTCONDOR/issues/HTCONDOR-1590)
- [HTCONDOR-1594](https://opensciencegrid.atlassian.net/jira/software/c/projects/HTCONDOR/issues/HTCONDOR-1594)

File: community/modules/scheduler/htcondor-configure/README.md
Comment (nick-stroud) [inline community/modules/scheduler/htcondor-configure/README.md:90]: nit: What is the value of commenting these lines out? I would maybe suggest something like:
```
vars:
...
  zone_secondary: a zone in same region as vars.zone
```

**Review by nick-stroud — APPROVED** [review]

File: community/modules/scheduler/htcondor-configure/README.md
Comment (tpdownes) [reply community/modules/scheduler/htcondor-configure/README.md]: Made change based on offline discussion.

**Pull request merged**
